### PR TITLE
Support for Let's Encrypt (ACME) certificates

### PR DESCRIPTION
**Pull request by stalehd**

### CODE REVIEW

File: restapi/server.go [hunk 1]
```diff
@@ -61,9 +64,24 @@ func NewServer(loopbackOnly bool, scontext *server.Context, config *server.Confi
...
+
+	if config.ACMECert {
+		logging.Info("Using Let's Encrypt for certificates")
+		m := &autocert.Manager{
+			Cache:      autocert.DirCache(config.ACMESecretDir),
+			Prompt:     autocert.AcceptTOS,
+			HostPolicy: autocert.HostWhitelist(config.ACMEHost),
+		}
+
+		go http.ListenAndServe(":http", m.HTTPHandler(nil))
```
Comment (gordonklaus) [inline restapi/server.go]: I had to look at the autocert documentation to understand what this does.  Probably deserves a comment.

Handy, though 👍

File: restapi/server.go [hunk 2]
```diff
@@ -25,6 +27,7 @@ import (
 	"github.com/ExploratoryEngineering/congress/server"
 	"github.com/ExploratoryEngineering/congress/utils"
 	"github.com/telenordigital/goconnect"
+	"golang.org/x/crypto/acme/autocert"
```
Comment (gordonklaus) [inline restapi/server.go]: newline 😉

**Review by gordonklaus — APPROVED** [review]
LGTM with a couple comments

**Review by hansj66 — APPROVED** [review]

File: restapi/server.go (hunk 1 above)
Comment (stalehd) [reply restapi/server.go]: It pulls up a simple http server to (I believe) the initial cert bootstrapping. Nice bonus is http->https redirects. This is basically a copy and paste from the examples.

Comment (gordonklaus) [reply restapi/server.go]: It only does the redirect.  It was added only recently (I had to update my dependency to use it in office canary).

**Pull request merged**